### DISCUSSION TITLE
Can't use an imported string array as parameter for `z.enum()`

### DISCUSSION THREAD

**Discussion by regisfaria** [colinhacks/zod#1976] [Q&A] [answered]
Hi! 

I'm my applications I like to create constant files which will be use a lot between files, keeping easier to maintain. 

In a project, I have the following constant:

```typescript
// in constants.ts
export const allowedLanguages = ['de', 'fr', 'en']
```

And when I tried to use that constant as parameter, to validate a contract using `zod.enum()` I have the following error:
<img width="264" alt="image" src="https://user-images.githubusercontent.com/44659603/216356558-297fe9b9-447c-4f37-92a8-207b17592fea.png">

<img width="978" alt="image" src="https://user-images.githubusercontent.com/44659603/216356511-0e61af90-3142-4c9c-af10-6648fc56ed01.png">

Is this a bug or intended? Why can't I use an exported array here? Feels bad having to edit all my schemas that will use language, as my supported languages grows...

Thanks for the time, and I'd be happy to help to allow this if possible ;)

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?
https://github.com/colinhacks/zod#zod-enums

```ts
const array = [ 'foo', 'bar' ]
z.enum( array ) // compile error

const readOnlyArray = [ 'foo', 'bar' ] as const
z.enum( readOnlyArray ) // no compile error
```

**Comment by regisfaria** [comment]
I missed that! Exactly what I was looking for! 

Thank you so much!

**Comment by JacobWeisenburger** [comment]
If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by genu** [comment]
I have found that this solution doesn't work if you have an object array and map it to a `string[]`

e.g.

```ts
const arr = [ {...},{...}].map(item=>item.value); // flat array of string[]

const readOnlyArray = arr as const // compile error

z.enum( readOnlyArray ) // compile error
```
